tools/wasme/cli/pkg/deploy/istio/istio_provider.go

```diff
@@ -105,25 +109,26 @@ type Provider struct {
 	WaitForCacheTimeout time.Duration
 }
 
-func NewProvider(ctx context.Context, kubeClient kubernetes.Interface, client ezkube.Ensurer, puller pull.ImagePuller, workload Workload, cache Cache, parentObject ezkube.Object, onWorkload func(workloadMeta metav1.ObjectMeta, err error), istioNamespace string, cacheTimeout time.Duration, ignoreVersionCheck bool) (*Provider, error) {
+func NewProvider(ctx context.Context, kubeClient kubernetes.Interface, client ezkube.Ensurer, puller pull.ImagePuller, workload Workload, cache Cache, parentObject ezkube.Object, onWorkload func(workloadMeta metav1.ObjectMeta, err error), istioNamespace, IstiodDeploymentName string, cacheTimeout time.Duration, ignoreVersionCheck bool) (*Provider, error) {
```

Comment:
Small nit: can we lowercase `IstiodDeploymentName` since it's local in scope here?
```suggestion
func NewProvider(ctx context.Context, kubeClient kubernetes.Interface, client ezkube.Ensurer, puller pull.ImagePuller, workload Workload, cache Cache, parentObject ezkube.Object, onWorkload func(workloadMeta metav1.ObjectMeta, err error), istioNamespace, istiodDeploymentName string, cacheTimeout time.Duration, ignoreVersionCheck bool) (*Provider, error) {
```

Reply:
done